core/src/main/java/com/linecorp/armeria/client/retry/MathUtils.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2017 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.client.retry;
+
+final class MathUtils {
+    private MathUtils() {}
+
+    static long safeAdd(long left, long right) {
+        if (left > Long.MAX_VALUE - right) {
```

Comment:
I may just be really silly, but I don't see how these safe methods will ever return Long.MAX_VALUE when the arguments are positive. How does this code work?

Anyways Guava has LongMath.saturatedAdd/Multiply